apps/evm/src/clients/api/queries/getMerklRewards/formatMerklRewardsResponse.ts

```diff
@@ -0,0 +1,139 @@
+import BigNumber from 'bignumber.js';
+import type { MerklDistribution, Token } from 'types';
+import { areAddressesEqual } from 'utilities';
+import type { GetMerklUserRewardsResponse, MerklRewardBreakdown } from '.';
+import type { PendingExternalRewardSummary } from '../getPendingRewards/types';
+
+type MerklCampaign = MerklDistribution & { poolComptrollerAddress: string };
+
+export const formatMerklRewardsPayload = (
+  apiPayload: GetMerklUserRewardsResponse,
+  merklCampaignsPerPool: Record<string, Record<string, MerklDistribution[]>>,
+): PendingExternalRewardSummary[] => {
+  const allCampaigns = Object.keys(merklCampaignsPerPool).reduce<MerklCampaign[]>(
+    (acc, poolComptrollerAddress) => {
+      const merklDistributionsPerAsset = merklCampaignsPerPool[poolComptrollerAddress];
+      const assetDistributionArray = Object.keys(merklDistributionsPerAsset).reduce<
+        MerklCampaign[]
+      >(
+        (accPoolCampaigns, vTokenAddress) => [
+          ...accPoolCampaigns,
+          ...merklDistributionsPerAsset[vTokenAddress].map(d => ({
+            ...d,
+            poolComptrollerAddress,
+          })),
+        ],
+        [],
+      );
+      return [...acc, ...assetDistributionArray];
+    },
+    [],
+  );
```

Comment:
If you don't need the `poolComptroller` prop, then this logic can be simplified:
```suggestion
  const allDistributions = Object.values(merklCampaignsPerPool).flatMap(
    merklDistributionsPerAsset =>
      Object.values(merklDistributionsPerAsset)
        .flat()
```